### PR TITLE
feat support excludeReferrers regex

### PR DESCRIPTION
**Pull request by Mercy811**
## UI
  - Keep the name of the existing textfield as `attributionExcludeReferrers` for backward compatibility
  - Add a textfield with name `attributionExcludeReferrersRegex` to support regex exclude referrers. 
#### Before
image

#### After
image

## Code 
Update sdk wrapper version to 3.9.0 https://github.com/amplitude/amplitude-js-gtm/pull/51

#### `generateConfiguration()` 
  - Assign `data.attributionExcludeReferrers` to `initOptions.defaultTracking.attribution.excludeReferrersText`
  - Assign `data.attributionExcludeReferrersRegex` to `initOptions.defaultTracking.attribution.excludeReferrersRegex`
  - The two exclude referrers in `initOptions` will be merged in https://github.com/amplitude/amplitude-js-gtm/pull/51

#### Test
- Add tests for init tag when exclude referrers text fields are set

### CODE REVIEW

File: template.tpl [hunk 1]
```diff
@@ -810,11 +810,24 @@ ___TEMPLATE_PARAMETERS___
...
-                        "simpleValueType": true,
-                        "help": "Pass a comma-separated list of referring domains you want to exclude from campaign attribution. \u003ca href\u003d\"\"\u003eRead more\u003c/a\u003e."
+                        "type": "GROUP",
+                        "name": "attributionExcludeReferrersGroup",
+                        "displayName": "",
+                        "groupStyle": "NO_ZIPPY",
+                        "subParams": [
+                          {
+                            "type": "TEXT",
+                            "name": "attributionExcludeReferrersText",
```
Comment (izaaz) [inline template.tpl:820]: You're changing the name of the field. When the customer updates the GTM plugin, they'll lose the historic value for this field?

Comment (Mercy811) [reply template.tpl]: Good call out. I updated to keep the name unchanged for backward compatibility.

File: template.tpl [hunk 2]
```diff
@@ -1357,11 +1364,15 @@ const generateConfiguration = () => {
 
     if (!!data.detAttribution) {
       initOptions.defaultTracking.attribution = {};
-
+
```
Comment (yuhao900914) [inline template.tpl:1367]: Is this intentional?

**Comment by yuhao900914** [comment]
Can we add the test to cover to cover this?

File: template.tpl (hunk 2 above)
Comment (Mercy811) [reply template.tpl:1367]: I added some logs and deleted them after debugging. These indents are added by VSCode. Not sure why they were not there before.

**Comment by Mercy811** [comment]
Hi @yuhao900914, good call out, I added tests in the latest commit.

File: template.tpl [hunk 3]
```diff
@@ -1680,6 +1680,205 @@ ___WEB_PERMISSIONS___
...
+    });
+
+    // Call runCode to run the template's code.
+    runCode(mockData);
+
+    // Verify that the tag finished successfully.
+    assertApi('gtmOnSuccess').wasCalled();
+- name: Init tag with single exclude referrers regex
+  code: |-
+    const excludeReferrersRegex = ["test.com", "text.org"];
```
Comment (yuhao900914) [inline template.tpl:1764]: This should be a single exclude referrers instead an array?

Comment (Mercy811) [reply template.tpl]: Nice catch! Updated

File: template.tpl [hunk 4]
```diff
@@ -1669,6 +1680,205 @@ ___WEB_PERMISSIONS___
...
+
+    // Test the following line:
+    // _amplitude(instanceName, 'init', data.apiKey, initUserId, generateConfiguration());
+    mock('copyFromWindow', key => {
+      return function() {
+        assertThat(arguments[0], 'Incorrect instance name').isEqualTo(mockData.instanceName);
+        assertThat(arguments[1], 'Incorrect tag type').isEqualTo(mockData.type);
+        assertThat(arguments[2], 'Incorrect apiKey object').isEqualTo(mockData.apiKey);
+        assertThat(arguments[3], 'Incorrect user Id').isEqualTo(null);
+         assertThat(arguments[4], 'Incorrect config').isEqualTo(expectedConfig);
```
Comment (yuhao900914) [inline template.tpl:1792]: Extra space here.

File: template.tpl [hunk 5]
```diff
@@ -1669,6 +1680,205 @@ ___WEB_PERMISSIONS___
...
```
Comment (yuhao900914) [inline template.tpl:1831]: Extra space.

**Review by yuhao900914 — APPROVED** [review]

File: template.tpl (hunk 4 above)
Comment (Mercy811) [reply template.tpl]: Updated at https://github.com/amplitude/amplitude-browser-sdk-gtm-template/pull/64/commits/a51a9241fb251c3649a4db950850f26157e73dda

File: template.tpl (hunk 5 above)
Comment (Mercy811) [reply template.tpl]: Updated at https://github.com/amplitude/amplitude-browser-sdk-gtm-template/pull/64/commits/a51a9241fb251c3649a4db950850f26157e73dda

**Pull request merged**